### DISCUSSION TITLE
How can nrich-registry be used with JWT authorization

### DISCUSSION THREAD

**Discussion by nkescec** [croz-ltd/nrich#148] [Q&A] [answered]
In some projects each individual route in a controller is annotated with `@PreAuthorize` which triggers some authorization based on JWTs.

Can controller routes exposed with nrich-registry use the same mechanism for authorization or is there a prefered way to implement it (for example, using spring security route authorization)?

**Comment by agrancaric** [comment, marked as answer]
I think the easiest way to do it would be with spring route authorization, alternatively if you need more fine grained control you can implement `net.croz.nrich.registry.api.data.interceptor.RegistryDataInterceptor` and do your checks there.
